lib/ulid/decoder.rb

```diff
@@ -0,0 +1,23 @@
+# frozen-string-literal: true
+
+require 'ulid/generator'
+
+module ULID
+  module Decoder
+    def decode_time(string)
+      if string.size != Generator::ENCODED_LENGTH
+        raise ArgumentError, 'string is not of ULID length'
+      end
+      epoch_time_ms = string.slice(0, 10).split('').reverse
+                            .each_with_index.reduce(0) do |carry, (char, index)|
+        encoding_index = Generator::ENCODING.index(char.bytes.first)
+        if encoding_index.nil?
+          raise ArgumentError, "invalid character found: #{char}"
+        end
+        carry += encoding_index * (Generator::ENCODING.size**index).to_i
+        carry
+      end
+      Time.at(epoch_time_ms / 1000, epoch_time_ms % 1000, :millisecond)
```

Comment:
1. `Time.at` with 3rd argument as unit is provided since Ruby 2.5.
So this change will drop to support Ruby 2.4 or older. Is this intentional? [GitHub Actions currently testing only in 2.6+](https://github.com/rafaelsales/ulid/blob/b30df17d10d1cd9330f40193d74bcd6d081a4b5b/.github/workflows/ruby.yml#L11). However [gemspec](https://github.com/rafaelsales/ulid/blob/b30df17d10d1cd9330f40193d74bcd6d081a4b5b/ulid.gemspec#L17-L20) looks to hope to care older rubies.

2. When using Time.at with unit, `msec / 1000` and `msec % 1000` calculation can be removed, I think.
  I use similar way as [this](https://github.com/kachick/ruby-ulid/blob/169867d4a55d4ec304c1b0d0ce458007a22ddd9d/lib/ulid.rb#L409).

```suggestion
      Time.at(0, epoch_time_ms, :millisecond)
```

Reply:
This seems correct and a nice simplification. We should definitely give it a shot